TW 64 Updated information about public collaboration (#3919)
postmanlabs/postman-docs: src/pages/docs/collaborating-in-postman/using-workspaces/public-workspaces.md

## Managing a public workspace

When a user opens a public workspace, Postman shows an overview of its contents, activity, and members in a new tab. The user interacts with the contents of the public workspace—the collections, APIs, environments, and other elements. As a team member with an Admin role on the workspace, you can manage the overview, the elements in the workspace, and the members of the workspace.

### Editing public workspace details

Users with an [Admin role](/docs/collaborating-in-postman/roles-and-permissions/#workspace-roles) for a workspace can edit workspace details, including updating the workspace's name, summary, description, and visibility. For more information, see [Editing workspace details](/docs/collaborating-in-postman/using-workspaces/managing-workspaces/#editing-workspace-details).

> The [Public API Network](/docs/getting-started/exploring-public-api-network/) displays your workspace's name and summary. Make your workspace name and summary informative to improve its visibility on the Public API Network.

### Adding elements to a public workspace

> You can add existing collections and environments to another public workspace by sharing them.

Reviewer: I don't think this is true anymore, as elements can only exist in a single workspace now. Sharing a collection/environment just shares a link but doesn't make it available in another workspace. To do that, you would have to move the collection/environment.

Author: You're absolutely right! I should have gone through the instructions in the existing doc much more thoroughly. Trust but verify!

* To share a _collection_ to a public workspace, open it using Collections on the left side, then select the three dots icon <img alt="Three dots icon" src="https://assets.postman.com/postman-docs/icon-three-dots-v9.jpg" width="18px" style="vertical-align:middle;margin-bottom:5px"> > **Share collection**.
* To share an _environment_ to a public workspace, open it using Environments on the left side, then select the three dots icon <img alt="Three dots icon" src="https://assets.postman.com/postman-docs/icon-three-dots-v9.jpg" width="18px" style="vertical-align:middle;margin-bottom:5px"> > **Share environment**.

Select the public workspace you want to share to, and choose whether you also want to remove the element from its current workspace. Select **Share and Continue**.

### Removing elements from a public workspace

> When you remove an element from a public workspace, it's no longer visible in that particular workspace. The element is still available in any other workspace in which it exists.

Reviewer: I think these instructions need to be updated. Elements like collections and environments can no longer be in multiple workspaces, and rather can only exist in a single workspace. You could either move the collection/environment or delete it. The "Remove from workspace" option  is not available under the three dots menu.

Author: Yikes, it's true, this is wildly out of date 🙈

* To remove an _environment_ from a public workspace, select the three dots icon <img alt="Three dots icon" src="https://assets.postman.com/postman-docs/icon-three-dots-v9.jpg" width="18px" style="vertical-align:middle;margin-bottom:5px"> next to the environment menu or select <img alt="Three dots icon" src="https://assets.postman.com/postman-docs/icon-three-dots-v9.jpg" width="18px" style="vertical-align:middle;margin-bottom:5px"> next to the Share button in the Environment tab. Select **Remove from workspace** and confirm that you would like to remove the environment from the current workspace. The environment will no longer be visible in the workspace.

> Deleting an element is different from removing it. When you delete an element, it's no longer available in any workspace. When you remove an element from a specific workspace, it will still be available in any other workspaces in which it exists.

### Deleting a public workspace

> To delete a public workspace, you must revert its visibility. If you try to delete a public workspace by selecting the three dots icon <img alt="Three dots icon" src="https://assets.postman.com/postman-docs/icon-three-dots-v9.jpg" width="18px" style="vertical-align:middle;margin-bottom:5px"> on the right corner of the workspace overview, the option to delete is greyed out.

Reviewer: Per Microsoft, change "is greyed out" to "Isn't available"
https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/g/gray-grayed-out

> ![Deleting a public workspace](https://assets.postman.com/postman-docs/delete-option-greyed-public-workspace.jpg)

Reviewer: Is this screenshot accurate? When I look at public workspaces, I don't see the three dots at upper right, but I do see "Workspace Settings"

Author: Updating!

To change the visibility of a public workspace and delete it:

> 1. Change the visibility dropdown to **Team**, then **Save Changes**.

Reviewer: Is this still accurate? When I look at a public workspace, I have to select "Workspace Settings" at upper right and then there's a delete button at the bottom

Author: Yipes, you're right! The old instructions were not up to date and I neglected to verify 😬  Will update!

* **Assign access to elements within a workspace.** Workspace roles control access to elements. You can assign workspace roles to an individual user or to a user group. To learn more about assigning workspace roles or removing a user from a workspace, see [Managing workspace roles](/docs/collaborating-in-postman/using-workspaces/managing-workspaces/#managing-workspace-roles).

> Team members will receive an email and in-app notification when they're added to a public workspace.

## Collaborating in another user's public workspace

To collaborate with someone else in their public workspace, open the **Workspace** menu and select the workspace you want to work in.

For **collections** and **environments**, [create a fork](/docs/collaborating-in-postman/version-control-for-collections/#forking-a-collection) and request to merge changes using a [pull request](/docs/collaborating-in-postman/version-control-for-collections/#creating-pull-requests).

> For **APIs**, navigate to the API and version. Select **Definition** > **Request Access** to request an editor role.

Reviewer: ```suggestion
For **APIs**, navigate to the API and version. Select the **Definition** tab and then select **Request Access** to request an editor role.
```

Reviewer: Per Microsoft, we can use ">" to connect actions that use the same selection method. But in this case you're selecting a tab and then a button, so I think it's best not to use ">"
https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui

---
title: "Public workspaces"
order: 76
page_id: "public_workspaces"
updated: 2022-03-11
warning: false
contextual_links:
  - type: section
    name: "Additional Resources"
  - type: subtitle
    name: "Related Blog Posts"
  - type: link
    name: "Creating workspaces"

>     url: "/docs/collaborating-in-postman/using-workspaces/creating-workspaces/"

Author: Good call! Will do.

* [Managing a public workspace](#managing-a-public-workspace)
    * [Editing public workspace details](#editing-public-workspace-details)
    * [Moving elements to a public workspace](#moving-elements-to-a-public-workspace)
    * [Deleting elements from a public workspace](#deleting-elements-from-a-public-workspace)
    * [Deleting a public workspace](#deleting-a-public-workspace)
    * [Managing public workspace users and roles](#managing-public-workspace-users-and-roles)
* [Collaborating in another user's public workspace](#collaborating-in-another-users-public-workspace)
* [Next steps](#next-steps)

## Enabling your team profile

Before other Postman users can work with your team in a public workspace, your team needs to have a public profile. A public team profile will encourage others to collaborate with you in a public workspace, allow you to ask for contributions and gather feedback, and will help increase your API's discovery in search results. A public team profile will also show up on the [Public API Network](https://www.postman.com/explore).

To make your [team profile](https://go.postman.co/settings/team/general) public:

> 1. Select **Team** next to your avatar in the upper-right corner of Postman, then select **Team settings**.

Reviewer 2: Could consider using a > (mentioned [here](https://github.com/postmanlabs/postman-docs/blob/develop/DOCS_STYLE_GUIDE.md))

<img alt="Enable team profile" src="https://assets.postman.com/postman-docs/edit-team-profile-v9.1.0.jpg"/>

For more tips on how to create a public profile, see [How to create your Postman profile](https://blog.postman.com/how-to-create-your-postman-public-profile/) on the Postman blog.

## Creating a public workspace

To create a new public workspace:

1. Open the workspace menu, then select **Create Workspace**.

    <img alt="Workspace menu" src="https://assets.postman.com/postman-docs/workspace-switcher-v9.1.jpg" width="400px"/>

1. Specify a workspace **Name** and **Summary**.
1. Under **Visibility**, select **Public**.
1. Select **Create workspace**.

    <img alt="Create public workspace" src="https://assets.postman.com/postman-docs/create-public-wkspc-v9.5.jpg" width="400px"/>

> Within a team, you can't have two public workspaces with the same name.

To convert an existing workspace to a public workspace:

1. Open the workspace. In the **Overview** tab, select **Workspace Settings** in the upper right.
1. Under **Visibility**, select **Public**.
1. Select **Update**.

> > You must have an [Admin](/docs/collaborating-in-postman/roles-and-permissions/#workspace-roles) role for a workspace to change its visibility to public.

Reviewer 2: not sure on caps, but maybe something like

```suggestion
> You must be a [Workspace Admin](/docs/collaborating-in-postman/roles-and-permissions/#workspace-roles) to change the visibility of a workspace.
```
